### PR TITLE
Test for verification of excluding unresponsive MN from MNList

### PR DESCRIPTION
**Pull request by abvgedeika**

### CODE REVIEW

File: test/src/MNDiscovery/index.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ const {expect} = chai;
 
 const MockedMNList = SMNListFixture.getFirstDiff();
 
-const masternodeIps = SMNListFixture.getFirstDiff().mnList.map(masternode => masternode.service.split(':')[0]);
+const masternodeIps = SMNListFixture.getFirstDiff().mnList.map(masternode => masternode.service.substring(0, masternode.service.lastIndexOf(":")).replace("[", "").replace("]", ""));
```
Comment (shumkov) [inline test/src/MNDiscovery/index.js]: Let's make it more readable? For example, we could move `masternode.service.lastIndexOf(":")).replace("[", "").replace("]", "")` to a variable first?

File: test/src/MNDiscovery/index.js [hunk 2]
```diff
@@ -114,6 +114,40 @@ describe('MNDiscovery', async () => {
...
+                    return ips.indexOf(elem) == pos;
+                });
+                for (const ip of uniqueIps) {
+                    //excludeIps should not contain random master nodes
+                    expect(excludeIps.indexOf(ip)).to.equal(-1);
+                }
+                discovery.masternodeListProvider.getMNList.restore();
+            }
+
+            await verifyRandomf([0, 1, 2, 3]);
```
Comment (shumkov) [inline test/src/MNDiscovery/index.js]: Why did we place it in the function?

Comment (abvgedeika) [reply test/src/MNDiscovery/index.js]: updated

File: test/src/MNDiscovery/index.js (hunk 1 above)
Comment (abvgedeika) [reply test/src/MNDiscovery/index.js]: modified

File: test/src/MNDiscovery/index.js [hunk 3]
```diff
@@ -91,28 +96,54 @@ describe('MNDiscovery', async () => {
...
+            let uniqueIps = ips.filter(function (elem, pos) {
+                return ips.indexOf(elem) == pos;
+            });
+            expect(uniqueIps.length > 1).to.be.true;
+            discovery.masternodeListProvider.getMNList.restore();
+        });
+
+        it('Should excludedIps from MN list', async () => {
+            const discovery = new MNDiscovery();
+            sinon.spy(discovery.masternodeListProvider, 'getMNList');
```
Comment (shumkov) [inline test/src/MNDiscovery/index.js]: Is it a unit test or functional? We will make a real request if `needsUpdate` is `true`.

Comment (abvgedeika) [reply test/src/MNDiscovery/index.js]: I would call them functional tests. I've added new test that verify that getRandomMasternode take into account excludedIP even when needsUpdate returns true

**Comment by shumkov** [comment]
Obsolete after #140

**Pull request closed without merging**